### PR TITLE
Featured content: fail gracefully when taxs are filtered out

### PR DESCRIPTION
**Pull request by jeherve**
#### Changes proposed in this Pull Request:

In some scenarios, plugins or themes using the get_terms filter may not return the array of taxonomies we expect in Featured Content.
In those situations, let's fail gracefully to avoid notices such as:

```
Warning: in_array() expects parameter 2 to be array, null given in wp-content/plugins/jetpack/modules/theme-tools/featured-content.php on line 394
```

This was originally reported in the forums:
https://wordpress.org/support/topic/after-updating-to-wordpress-5-7-getting-jetpack-error/

#### Jetpack product discussion

* N/A

#### Does this pull request change what data or activity we track or use?

* No

#### Testing instructions:

Not too sure how to reproduce yet, you would need your own snippet to filter `get_terms` but not take the second parameter into account properly.
To ensure Featured Content still works, try the following:

* Set up a new site with a theme like https://wordpress.org/themes/superhero/
* Publish a few posts with large featured images
* Go to Appearance > Customize and check the Featured Content settings
* Make sure they still work, that you can exclude a post if it has a specific tag).

### CODE REVIEW

**Review by sdixon194 — APPROVED** [review]
This seems fairly logical to me. I tried to reproduce the error by setting up a filter in a snippet and playing around, but couldn't reproduce it as well.

File: projects/plugins/jetpack/modules/theme-tools/featured-content.php [hunk 1]
```diff
@@ -390,6 +390,10 @@ public static function hide_featured_term( $terms, $taxonomies, $args ) {
 				return $terms;
 			}
 
+			if ( empty( $taxonomies ) ) {
```
Comment (anomiex) [inline projects/plugins/jetpack/modules/theme-tools/featured-content.php:393]: Feel free to ignore, unnecessary use of `empty()` is one of my pet peeves.
```suggestion
			// WordPress defines the parameter as `array`, but it passes null if `get_terms( $args )` was called
			// without a 'taxonomy' in $args.
			if ( ! is_array( $taxonomies ) ) {
```

**Review by anomiex — COMMENTED** [review]
Took a look at the code. The filter is called from WordPress's `get_terms()`, which since WP 4.5's new calling convention can be called without specifying a taxonomy at all. In this case, despite the documentation stating that the filter will be passed an array for `$taxonomies`, it gets passed null instead of an empty array. At a quick glance WP itself seems to treat null as empty array.

I saw no stack trace in the original report, but I'd guess some other plugin or theme (rather than WP itself) is calling `get_terms()` in that manner.

File: projects/plugins/jetpack/changelog/fix-error-featured-content-tax [hunk 2]
```diff
@@ -0,0 +1,4 @@
+Significance: patch
+Type: compat
+
+Featured Content: avoid PHP warnings when another plugin or theme filters out taxonomies.
```
Comment (anomiex) [inline projects/plugins/jetpack/changelog/fix-error-featured-content-tax:4]:
```suggestion
Featured Content: avoid PHP warnings when terms are fetched without a taxonomy to filter by.
```

**Review by anomiex — COMMENTED** [review]
I should have thought to update the changelog entry earlier. Sorry.

**Comment by jeherve** [comment]
r224751-wpcom

**Review by anomiex — APPROVED** [review]

**Pull request merged**